modules/gdscript/gdscript_cache.cpp

```diff
@@ -133,7 +135,10 @@ Ref<GDScriptParserRef> GDScriptCache::get_parser(const String &p_path, GDScriptP
 		ref->path = p_path;
 		singleton->parser_map[p_path] = ref.ptr();
 	}
-
+	if (ref.is_null()) {
+		r_error = ERR_INVALID_DATA;
+		return nullptr;
```

Comment:
Earlier error checks return `ref` (which is `Ref<GDScriptParserRef>()`), this should likely do the same for consistency.

Or actually, the `else` above doesn't look like it can have an invalid `ref` (it's instantiated there), so this error check should be done in:
```
if (singleton->parser_map.has(p_path)) {
		ref = Ref<GDScriptParserRef>(singleton->parser_map[p_path]);
		if (ref.is_null()) {
			r_error = ERR_INVALID_DATA;
			return ref;
		}
} else {
```